PWGHF/Core/HfMlResponseXic0ToXiPi.h

```diff
@@ -0,0 +1,128 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file HfMlResponseXic0ToXiPi.h
+/// \brief Class to compute the ML response for Ξc^0 → Ξ∓ π± analysis selections
+/// \author Tao Fang <tao.fang@cern.ch>, Central China Normal University
+
+#ifndef PWGHF_CORE_HFMLRESPONSEXIC0TOXIPI_H_
+#define PWGHF_CORE_HFMLRESPONSEXIC0TOXIPI_H_
+
+#include "PWGHF/Core/HfMlResponse.h"
+
+#include "Tools/ML/MlResponse.h"
+
+#include <cstdint>
+#include <vector>
+
+// Fill the map of available input features
+// the key is the feature's name (std::string)
+// the value is the corresponding value in EnumInputFeatures
+#define FILL_MAP_XIC0TOXIPI(FEATURE)                                 \
+  {                                                                  \
+    #FEATURE, static_cast<uint8_t>(InputFeaturesXic0ToXiPi::FEATURE) \
+  }
+
+// Check if the index of mCachedIndices (index associated to a FEATURE)
+// matches the entry in EnumInputFeatures associated to this FEATURE
+// if so, the inputFeatures vector is filled with the FEATURE's value
+// by calling the corresponding GETTER from OBJECT
+#define CHECK_AND_FILL_VEC_XIC0TOXIPI_FULL(OBJECT, FEATURE, GETTER) \
+  case static_cast<uint8_t>(InputFeaturesXic0ToXiPi::FEATURE): {    \
+    inputFeatures.emplace_back(OBJECT.GETTER());                    \
+    break;                                                          \
+  }
+
+// where OBJECT is named candidate and FEATURE = GETTER
+#define CHECK_AND_FILL_VEC_XIC0TOXIPI(GETTER)                   \
+  case static_cast<uint8_t>(InputFeaturesXic0ToXiPi::GETTER): { \
+    inputFeatures.emplace_back(candidate.GETTER());             \
+    break;                                                      \
+  }
+
+namespace o2::analysis
+{
+
+enum class InputFeaturesXic0ToXiPi : uint8_t {
+  tpcNSigmaPiFromLambda,
+  tpcNSigmaPiFromCasc,
+  tpcNSigmaPiFromCharmBaryon,
+  dcaCascDau,
+  dcaCharmBaryonDau,
+  cosPACharmBaryon,
+  cosPAV0,
+  impactParBachFromCharmBaryonXY,
+  impactParCascXY
```

Comment:
Are these the only needed features? For instance, why only the TPC PID for the pion from Lambda, and not the proton from Lambda? Why only the cosPA of the V0 and not the Xi cascade?